src/main/java/software/amazon/smithy/lsp/diagnostics/SmithyDiagnostics.java

```diff
@@ -208,8 +208,19 @@ default Diagnostic toDiagnostic(ValidationEvent event) {
                 default -> DiagnosticSeverity.Hint;
             };
             var diagnosticRange = getDiagnosticRange(event);
-            var message = event.getId() + ": " + event.getMessage();
-            return new Diagnostic(diagnosticRange, message, diagnosticSeverity, "Smithy");
+
+            var message =
+                new StringBuilder(event.getId())
+                    .append(": ")
+                    .append(event.getMessage());
+
+            event.getHint().ifPresent(hint ->
+                message
+                    .append("\n\nHint: ")
+                    .append(hint)
+            );
+
+            return new Diagnostic(diagnosticRange, message.toString(), diagnosticSeverity, "Smithy");
         }
```

Comment:
```suggestion
            var message = getMessage(event);
            return new Diagnostic(diagnosticRange, message, diagnosticSeverity, "Smithy");
        }
       
       String HINT_PREFIX = System.lineSeparator() + System.lineSeparator() + "Hint: ";

        private static String getMessage(ValidationEvent event) {
            var hint = event.getHint().orElse(null);
            if (hint == null) {
                return event.getId() + ": " + event.getMessage();
            }

            return event.getId() + ": " + event.getMessage() + HINT_PREFIX + hint;
        }
```
I think the VM is better at regular string concatenations these days, outside of a loop